File: ts/src/external/conversion-rate-api.ts
/**
 * This class is provided by a third party and you cannot change it.
 */
import { CurrencyIsoCode } from "./currency-iso-code";

export class ConversionRateApi {
  getRate(source: CurrencyIsoCode, target: CurrencyIsoCode): number {
    console.log(`Getting rate from ${source} to ${target}`);
    throw new Error(
      "401 - Unauthorized, IP not recognized. Are you trying to call us via a unit test?"
    );
  }
}

File: java/src/main/java/external/ConversionRateApi.java
package external;

public class ConversionRateApi {
    public double getRate(CurrencyIsoCode source, CurrencyIsoCode target) {
        System.out.println("Getting rate from " + source + " to " + target);
        throw new RuntimeException("401 - Unauthorized, IP not recognized. Are you trying to call us via a unit test?");
    }
}
